lambda/translcore.ml

```diff
@@ -724,22 +724,24 @@ and transl_apply ~scopes ?(should_be_tailcall=false) ?(inlined = Default_inline)
      : Lambda.lambda)
 
 and transl_function0
-      ~scopes loc return untuplify_fn repr partial (param:Ident.t) cases =
+      ~scopes loc return untuplify_fn max_arity
+      repr partial (param:Ident.t) cases =
   match cases with
     [{c_lhs=pat; c_guard=None;
       c_rhs={exp_desc = Texp_function { arg_label = _; param = param'; cases;
         partial = partial'; }; exp_env; exp_type} as exp}]
-    when Parmatch.inactive ~partial pat ->
+    when max_arity > 1 && Parmatch.inactive ~partial pat ->
       let kind = value_kind pat.pat_env pat.pat_type in
       let return_kind = function_return_value_kind exp_env exp_type in
       let ((_, params, return), body) =
-        transl_function0 ~scopes exp.exp_loc return_kind false
+        transl_function0 ~scopes exp.exp_loc return_kind false (max_arity - 1)
           repr partial' param' cases
       in
       ((Curried, (param, kind) :: params, return),
        Matching.for_function ~scopes loc None (Lvar param)
          [pat, body] partial)
-  | {c_lhs={pat_desc = Tpat_tuple pl}} :: _ when untuplify_fn ->
+  | {c_lhs={pat_desc = Tpat_tuple pl}} :: _
+    when untuplify_fn && List.length pl <= max_arity ->
```

Comment:
@mshinwell: I think this ensures that tuplified functions that reach Flambda always have an arity small enough that generating the worker function works. So I believe no patch on flambda is required. Does that seem correct to you ?

Reply:
Ah yes, indeed -- I think that claim is correct and I was wrong before.  So we shouldn't need to do anything in Flambda.